.ado/templates/integration-test-job.yml

```diff
@@ -0,0 +1,113 @@
+      - script: yarn windows --no-launch --no-packager --no-deploy --no-autolink --arch ${{ parameters.BuildPlatform }} --logging --buildLogDirectory $(BuildLogDirectory)
+        displayName: yarn windows --no-launch
+        workingDirectory: packages/IntegrationTest
+
+      # This will normally be run during tests if it isn't already, but running
+      # it before helps avoid our window from being obscured.
+      - powershell: Start-Process CheckNetIsolation.exe -ArgumentList "LoopbackExempt","-is" -Verb runAs
+        displayName: Exempt loopback server
+
+      - powershell: Start-Process npm -ArgumentList "run","start"
+        displayName: Start packager
+        workingDirectory: packages/IntegrationTest
+
+      # Instance load may time out if the packager takes too long to return a
+      # bundle. Request the bundle ahead of time so that the instance will get
+      # it back quickly.
+      - powershell: |
+          while ($true) {
+            try {
+              Invoke-WebRequest -Uri "http://localhost:8081/index.bundle?platform=windows&dev=true"
+              break
+            } catch [Exception] {
+              echo $_.Exception
+              Start-Sleep -s 1
+            }
+          }
+        displayName: Warm packager
+
+      # If a debugger isn't connected to the packager, the CLI will launch the
+      # default browser to go there. On Windows Server that means launching IE
+      # with a bunch of restrictions. Explicitly launch Chrome to the debugger
+      # UI instead.
+      - powershell: Start-Process chrome http://localhost:8081/debugger-ui/
+        displayName: Launch debugger-ui
+
+      # There's a slight race condition here, where we assume Chrome will open
+      # before this launches the app. This step takes ~1m on CI machines so it
+      # shouldn't be a practical concern.
+      - script: yarn windows --no-build --no-packager --no-autolink --arch ${{ parameters.BuildPlatform }} --logging
+        displayName: yarn windows --no-build
+        workingDirectory: packages/IntegrationTest
+
+      - powershell: |
+          $wshell = New-Object -ComObject wscript.shell
+          $wshell.AppActivate('integrationtest')
+        displayName: Activate test window
+
+      # Wait a brief moment before taking a screenshot to give the bundle a chance to load
+      - powershell: Start-Sleep -s 5
+        displayName: Pause before screenshot
+
+      - script: yarn take-screenshot --filename $(Build.StagingDirectory)/Screenshots/before-test.png
+        displayName: Screenshot desktop before tests
+        condition: succeededOrFailed()
+
+      - script: yarn integration-test
+        displayName: yarn integration-test
+        workingDirectory: packages/IntegrationTest
+
+      - script: yarn take-screenshot --filename $(Build.StagingDirectory)/Screenshots/after-test.png
+        displayName: Screenshot desktop after tests
+        condition: succeededOrFailed()
+
+      - task: PublishPipelineArtifact@1
```

Comment:
Suggestion for future expansion:
* Log some memory metrics to catch any memory leaks we have.

Reply:
Yeah, I think perf/memory is something we should really be monitoring. For memory, Office actually has some stuff for low-overhead shutdown time leak detection that I think is really nice (especially compared to more generic methods) but it brings a lot with it, and even modifies some STL headers to get there. Might be interesting to see if we could do special leak detection runs though.

FWIW I think the biggest concern we have would be ref cycles, since we have a nice modern codebase using RAII everywhere.